extensions/wikia/Search/classes/Config.php

```diff
@@ -512,62 +588,63 @@ public function hasWikiMatch() {
 
 	/**
 	 * Stores the current article match ONLY IF IT PASSES OUR ESTABLISHED FILTERS
+	 *
 	 * @param  \Wikia\Search\Match\Article $articleMatch
-	 * @return \Wikia\Search\Config provides fluent interface
+	 *
+	 * @return Config provides fluent interface
 	 */
 	public function setArticleMatch( Match\Article $articleMatch ) {
 		if ( $this->articleMatchPassesFilters( $articleMatch ) ) {
 			$this->articleMatch = $articleMatch;
 		}
+
 		return $this;
 	}
 
 	/**
 	 * Here, we're checking for conditions that should preclude a match, given our current environment settings.
 	 * We're using DeMorgan's theorem here. So write FOR the condition you're trying to filter out.
+	 *
 	 * @param \Wikia\Search\Match\Article $match
+	 *
 	 * @return boolean
 	 */
 	protected function articleMatchPassesFilters( \Wikia\Search\Match\Article $match ) {
 		$result = $match->getResult();
 		$filterKeys = $this->getPublicFilterKeys();
 		$isVideoFile = $this->getService()->pageIdIsVideoFile( $result['pageid'] );
 		$minArticleQuality = $this->getMinArticleQuality();
-		if($minArticleQuality && $result['article_quality_i'] <= $minArticleQuality) return false;
-
-		return ! (
-				(
-					$minArticleQuality
-					&&
-					( $result['article_quality_i'] < $minArticleQuality )
-				) || ( // We have a file that is video, but we only want images.
-						$result['ns'] == NS_FILE
-						&&
-						in_array( \Wikia\Search\Config::FILTER_IMAGE, $filterKeys )
-						&&
-						$isVideoFile
-				) || ( // We have a file that is not a video, but we only want videos.
-						$result['ns'] == NS_FILE
-						&&
-						in_array( \Wikia\Search\Config::FILTER_VIDEO, $filterKeys )
-						&&
-						!$isVideoFile
-				)
-		);
+		if ( $minArticleQuality && $result['article_quality_i'] <= $minArticleQuality ) {
+			return false;
+		}
+
+		return !( ( $minArticleQuality && ( $result['article_quality_i'] < $minArticleQuality ) ) ||
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (7) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)